marshal/proto_test.go

```diff
@@ -0,0 +1,142 @@
+package marshal
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/xitongsys/parquet-go/schema"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type subElem struct {
+	Val string
+}
+
+type testNestedElem struct {
+	SubElem     subElem
+	SubPtr      *subElem
+	SubList     []subElem
+	SubRepeated []*subElem
+	// Marshal ok but not write since it doesn't have corresponding primitive type
+	EmptyElem     struct{}
+	EmptyPtr      *struct{}
+	EmptyList     []struct{}
+	EmptyRepeated []*struct{}
+}
+type JobStatus int32
+
+const (
+	JobStatus_JobStatus_UNSPECIFIED  JobStatus = 0
+	JobStatus_BLOCKED                JobStatus = 1
+	JobStatus_ENQUEUED               JobStatus = 2
+	JobStatus_RUNNING                JobStatus = 3
+	JobStatus_COMPLETED              JobStatus = 4
+	JobStatus_ERRORED                JobStatus = 5
+	JobStatus_CANCELLED              JobStatus = 6
+	JobStatus_UPSTREAM_NOT_PROCESSED JobStatus = 7
+)
+
+func (x JobStatus) Enum() *JobStatus {
+	p := new(JobStatus)
+	*p = x
+	return p
+}
+
+func (x JobStatus) String() string {
+	statusToName := map[int32]string{
+		0: "ASYNCJOBSTATUS_UNSPECIFIED",
+		1: "BLOCKED",
+		2: "ENQUEUED",
+		3: "RUNNING",
+		4: "COMPLETED",
+		5: "ERRORED",
+		6: "CANCELLED",
+		7: "UPSTREAM_NOT_PROCESSED",
+	}
+	return statusToName[int32(x)]
+}
+
+type ProtoMessage struct {
+	Timestamp timestamppb.Timestamp
+	Status    JobStatus
+	IntVal    int32
```

Comment:
Do you think it's worth having another ProtoMessage/Struct as one Field here to also test **nested** objects?

Reply:
We have nested struct in another unit test. This one is more specifically to test the proto special handling for the timestamp and enum